services/postgres/UserTopicService.js

```diff
@@ -9,29 +9,43 @@ class UserTopicService {
     this._userTopicHistoryModel = UserTopicHistoryModel;
     this.followTopic = this.followTopic.bind(this);
     this.getUserTopic = this.getUserTopic.bind(this);
+    this.getFollowTopicStatus = this.getFollowTopicStatus.bind(this);
   }
 
-  async getFollowTopic(user_id, topic_id) {
-    try {
-    } catch (error) {}
-  }
-
-  async followTopic(user_id, topic_id) {
+  async getFollowTopicStatus(user_id, topic_id) {
     try {
       let topicUser = await this.getUserTopic(user_id, topic_id);
       if (topicUser) {
-        await this.deleteUserTopic(user_id, topic_id);
         return true;
       } else {
-        await this.addUserTopic(user_id, topic_id);
         return false;
       }
+    } catch (error) {
+      console.log(error);
+      throw new InvariantError('error get follow topic status');
+    }
+  }
+
+  async followTopic(user_id, topic_id) {
+    try {
+      await this.addUserTopic(user_id, topic_id);
+      return false;
     } catch (error) {
       console.log(error);
       throw new InvariantError('error follow topic');
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the `followTopic` method should include the original error message when throwing an `InvariantError`.

```diff
-       console.log(error);
-       throw new InvariantError('error follow topic');
+       throw new InvariantError(`Error following topic: ${error.message}`);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->